### PR TITLE
 Host Name is substituted with wrong value when server is configured only with bind attribute.

### PR DESCRIPTION
**Pull request by mobasherul**
For reference - 
https://github.com/Terracotta-OSS/terracotta-configuration/issues/96.

### CODE REVIEW

**Comment by mobasherul** [comment]
TDB-3333 for reference. ClusterTool configure command use to fail because we were not storing correct host address for tc-config in which server is configured with only bind attribute.

File: tc-config-parser/src/main/java/org/terracotta/config/TCConfigurationParser.java [hunk 1]
```diff
@@ -212,10 +212,12 @@ private static void initializeTsaGroupPort(Server server) {
 
   private static void initializeNameAndHost(Server server) {
     if (server.getHost() == null || server.getHost().trim().length() == 0) {
-      if (server.getName() == null) {
-        server.setHost("%i");
-      } else {
+      if (server.getName() != null) {
         server.setHost(server.getName());
+      } else if (server.getBind() != null) {
```
Comment (ramsai1729) [inline tc-config-parser/src/main/java/org/terracotta/config/TCConfigurationParser.java]: I think we need to make sure bind address is not a wildcard address as well.

File: tc-config-parser/src/main/java/org/terracotta/config/TCConfigurationParser.java [hunk 2]
```diff
@@ -212,10 +212,12 @@ private static void initializeTsaGroupPort(Server server) {
 
   private static void initializeNameAndHost(Server server) {
     if (server.getHost() == null || server.getHost().trim().length() == 0) {
-      if (server.getName() == null) {
-        server.setHost("%i");
-      } else {
+      if (server.getName() != null) {
         server.setHost(server.getName());
+      } else if (!server.getBind().equals(WILDCARD_IP)) {
```
Comment (ramsai1729) [inline tc-config-parser/src/main/java/org/terracotta/config/TCConfigurationParser.java]: Its better to use `WILDCARD_IP.equals(server.getBind())` even if `getBind()` can't be null.

**Comment by ramsai1729** [comment]
@myronkscott could you review this PR?

**Comment by mobasherul** [comment]
We don't need this change.

**Pull request closed without merging**